### PR TITLE
[ᚬframework] feat(framework): Service executor.

### PR DESCRIPTION
**Pull request by yejiayu**
**What type of PR is this?**
feature

**What this PR does / why we need it**:
The PR contains the following two parts:
- Service executor: used to execute the service specified in the transaction. see `framework/src /executor/mod.rs`
- AssetService: a build-in service through which you can create assets and transfer assets. see `built-in-services/asset/src/lib.rs`

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
The **dispatcher** module used for inter-service invocation is not yet implemented. @mkxbl

### CODE REVIEW

File: build-in-services/asset/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        ctx: Context,
+        payload: CreateAssetPayload,
+    ) -> ProtocolResult<Asset> {
+        let caller = ctx.get_caller();
+        let payload_str =
+            serde_json::to_string(&payload).map_err(|e| ServiceError::JsonParse(e))?;
+
+        let id = Hash::digest(Bytes::from(payload_str + &caller.as_hex()));
+
+        if self.assets.contains(&id)? {
```
Comment (zeroqn) [inline build-in-services/asset/src/lib.rs]: No symbol, name check?

File: build-in-services/asset/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            return Err(ServiceError::LackOfBalance {
+                expect: value,
+                real:   caller_balance,
+            }
+            .into());
+        }
+
+        let to_balance = self.sdk.get_account_value(&to, &asset_id)?.unwrap_or(0);
+        self.sdk
+            .set_account_value(&to, asset_id.clone(), to_balance + value)?;
```
Comment (zeroqn) [inline build-in-services/asset/src/lib.rs]: Should use ```checked_add``` or ```overflowing_add``` or ```saturating_add```.

File: build-in-services/asset/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        self.sdk
+            .set_account_value(&caller, asset_id.clone(), caller_balance - value)?;
```
Comment (zeroqn) [inline build-in-services/asset/src/lib.rs]: Same as above. ```checked_sub``` or ....

Comment (zeroqn) [reply build-in-services/asset/src/lib.rs]: IMO, we should sub first, if there's an error, we should handle it, restore user's balance.

Comment (yejiayu) [reply build-in-services/asset/src/lib.rs]: Any errors that occur during execution are rolled back.

File: build-in-services/asset/src/lib.rs (hunk 1 above)
Comment (yejiayu) [reply build-in-services/asset/src/lib.rs]: Id is generated by caller + name + symbol

**Comment by yejiayu** [comment]
ping @zeroqn

File: protocol/Cargo.toml [hunk 4]
```diff
@@ -12,19 +12,19 @@ derive_more = "0.15"
...
 lazy_static = "1.4"
 num-bigint = "0.2"
-ethbloom = "0.6"
+ethbloom = "0.8"
 bytes = "0.4"
 hex = "0.3"
 prost = "0.5"
 hasher = { version = "0.1", features = ['hash-keccak'] }
 creep = "0.1"
 bincode = "1.1"
```
Comment (KaoImin) [inline protocol/Cargo.toml]:
```suggestion
bincode = "1.2"
```

**Comment by yejiayu** [comment]
ping @KaoImin

**Pull request merged**